### ISSUE TITLE
Add commands to set alpha values directly

### ISSUE THREAD

**Issue by jpakkane** [pdf-association/pdf-issues#400]
I propose adding two new commands to the command stream:

- `a` to set alpha value for nonstroking operations
- `A` to set alpha value for stroking operations

I do know that the the PDF spec says that adding new commands is not expected. I do also know that you can set these both of these with a `gs` command. I even support the fact that most of the values in graphics state do not have their own setter commands.

However I do think that alpha channel should have it. The main reason for this is that they current state of things is cumbersome for a common use case.

Suppose you have a vector drawing program like Inkscape or Illustrator. A drawing in them may have hundreds of drawables that have been manually tuned by an artist using the color and transparency sliders in the drawing app. This means that there may be hundreds of different transparency levels in the picture. Exporting this faithfully means having a PDF graphics state dictionary object for every transparency level used. All of this manual work would go away if there was a command to set the alpha value directly.

Other entries in the graphics state change rarely so they do not need their own commands. But alpha should have it because setting alpha is a common operation and the values you can set it to can vary a lot within a single document.

**Comment by petervwyatt** [comment]
Will be considered as part of the [Image Model TWG](https://pdfa.org/community/imaging-model-twg/) so parking.
Please join if you wish to have your voice heard!

**Comment by stechio** [comment]
@jpakkane:
> I propose adding two new commands to the command stream:
> 
>     * `a` to set alpha value for nonstroking operations
> 
>     * `A` to set alpha value for stroking operations
> 
> [...]
> 
> Suppose you have a vector drawing program like Inkscape or Illustrator. A drawing in them may have hundreds of drawables that have been manually tuned by an artist using the color and transparency sliders in the drawing app. This means that there may be hundreds of different transparency levels in the picture. Exporting this faithfully means having a PDF graphics state dictionary object for every transparency level used. All of this manual work would go away if there was a command to set the alpha value directly.

That's a use case I encountered too, craving for those missing operators to match their coloring counterparts for stroking and nonstroking operations, for exactly the same reasons.

**Comment by jpakkane** [comment]
> Please join if you wish to have your voice heard!

As much as I'd like to I'm not paying [several hundred euros a month](https://pdfa.org/member-benefits/) for the privilege of being able to file bug reports.

**Comment by DuffJohnson** [comment]
> As much as I'd like to I'm not paying [several hundred euros a year](https://pdfa.org/member-benefits/) for the privilege of being able to file bug reports.

Bug reports are free - you didn't pay anything to file this one, did you? :-) 

But, if you want to participate in the conversation that results from your report, and have the opportunity to influence the outcome, this is why you'd attend the respective TWG meeting(s).

**Comment by jpakkane** [comment]
> Bug reports are free - you didn't pay anything to file this one, did you? :-)

No, but the answer was, basically, "this will be ignored, if you want something to happen, report it in this other place which does cost money".

The PDF Association seems to have been set up in a very "corporate-y" way for lack of a better term. I'm not a representative of any such organisations, just a random open source developer. I'm not earning any money out of this either now and probably will. The only free one is liaison, which is described with the text "Non-profit organizations use Liaison membership to engage with industry on advancing technologies to serve their interests.".  I am not an organization so this does not apply to me. Interestingly, as per the table on the page, liaison members are not even allowed to participate in actual working group meetings, only "liaison working groups" whatever that means.

This whole setup makes it infeasible for open source developers to contribute in any meaningful way. This need not be so. As a counterexample the ISO standardisation work on C++ is very open and anyone can participate in it.

If anyone with PDF Association membership reads this, feel free to copypaste the original report to your internal discussion forum (or whatever it is you use). I'll even be happy to answer any questions that arise on this bug. But unless there is a no-cost way for open source developers in general to contribute, that is all I have the capacity to do.

**Comment by DuffJohnson** [comment]
You misunderstood the answer. Your suggestion will be considered in the Imaging Model TWG with no further action on your part. Thank you for raising it!

The PDF Association is setup in a broadly similar way to, for example W3C; participation (at a certain level) requires paid membership. Numerous OS devs are already members (it's true that they typically have some sort of business interest in PDF technology).

The PDF Association does provide no-cost access to LWGs for interested parties and, like W3C, provides an Invited Expert program under which domain experts can request full Individual membership - including access to TWGs - at no cost. Obviously, as we also need to keep the lights on and the servers running, the Board of Directors limits the number of Invited Experts.

Please feel free to contact me offline if you have any specific questions regarding membership options.

**Comment by jpakkane** [comment]
> The PDF Association does provide no-cost access to LWGs for interested parties and, like W3C, provides an Invited Expert program under which domain experts can request full Individual membership - including access to TWGs - at no cost.

You might consider adding this information to [the membership type page](https://pdfa.org/member-benefits/).That is where most people would expect to find it, I'd wager.

**Comment by DuffJohnson** [comment]
> You might consider adding this information to [the membership type page](https://pdfa.org/member-benefits/).That is where most people would expect to find it, I'd wager.

We already discuss non-member access to LWGs on each of the LWG pages.  Good point re Invited Experts; I've updated that page to include that information. Thanks!
